contrib/startup_regtest.sh

```diff
@@ -142,17 +142,24 @@ start_ln() {
 	# Wait for it to start.
 	while ! bitcoin-cli -datadir="$PATH_TO_BITCOIN" -regtest ping 2> /tmp/null; do echo "awaiting bitcoind..." && sleep 1; done
 
-	# Kick it out of initialblockdownload if necessary
-	if bitcoin-cli -datadir="$PATH_TO_BITCOIN" -regtest getblockchaininfo | grep -q 'initialblockdownload.*true'; then
-		# Modern bitcoind needs createwallet
+	# Check if default wallet exists
+	if ! bitcoin-cli -datadir=$PATH_TO_BITCOIN -regtest listwalletdir | jq -r '.wallets[] | .name' | grep -wqe 'default' ; then
+		# wallet dir does not exist, create one
 		echo "Making \"default\" bitcoind wallet."
 		bitcoin-cli -datadir="$PATH_TO_BITCOIN" -regtest createwallet default >/dev/null 2>&1
-		# But it might already exist, load it
-	        bitcoin-cli -datadir="$PATH_TO_BITCOIN" -regtest loadwallet default
+	fi
+
+	# Check if default wallet is loaded
+	if ! bitcoin-cli -datadir=$PATH_TO_BITCOIN -regtest listwallets | jq -r '.[]' | grep -wqe 'default' ; then
```

Comment:
contrib/startup_regtest.sh:153:28: note: Double quote to prevent globbing and word splitting. [SC2086]